From ef171ea387e7a1d9204698dc3ea8ed663d5289b5 Mon Sep 17 00:00:00 2001
Message-ID: <ef171ea387e7a1d9204698dc3ea8ed663d5289b5.1740167398.git.mjg@fedoraproject.org>
From: Robin Watts <Robin.Watts@artifex.com>
Date: Fri, 21 Feb 2025 20:49:53 +0100
Subject: [PATCH] fix sys builds

---
 Makethird | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makethird b/Makethird
index e2cca8e5d..d4d4696b3 100644
--- a/Makethird
+++ b/Makethird
@@ -235,6 +235,7 @@ ifeq ($(HAVE_LEPTONICA),yes)
   THIRD_CFLAGS += -DHAVE_LEPTONICA
 ifeq ($(USE_SYSTEM_LEPTONICA),yes)
   THIRD_LIBS += $(SYS_LEPTONICA_LIBS)
+  LEPTONICA_CFLAGS += $(SYS_LEPTONICA_CFLAGS)
 else
   THIRD_LIBS += $(LEPTONICA_LIBS)
   THIRD_SRC += $(LEPTONICA_SRC)
@@ -249,6 +250,7 @@ ifeq ($(HAVE_TESSERACT),yes)
   THIRD_CFLAGS += -DHAVE_TESSERACT
 ifeq ($(USE_SYSTEM_TESSERACT),yes)
   THIRD_LIBS += $(SYS_TESSERACT_LIBS) -lstdc++
+  TESSERACT_CFLAGS += $(SYS_TESSERACT_CFLAGS)
 else
   THIRD_LIBS += $(TESSERACT_LIBS)
   THIRD_SRC += $(TESSERACT_SRC)
-- 
2.48.1.708.g19068738a8

