From c6bfa243c38a68b2a118efa2ea41b6c1ee5ff774 Mon Sep 17 00:00:00 2001
Message-ID: <c6bfa243c38a68b2a118efa2ea41b6c1ee5ff774.1760559226.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Wed, 15 Oct 2025 22:13:41 +0200
Subject: [PATCH] Fix fix for Bug 708845:

---
 source/pdf/pdf-image.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/pdf/pdf-image.c b/source/pdf/pdf-image.c
index d965fa824..d041b2a46 100644
--- a/source/pdf/pdf-image.c
+++ b/source/pdf/pdf-image.c
@@ -28,7 +28,7 @@
 static fz_image *pdf_load_jpx(fz_context *ctx, pdf_document *doc, pdf_obj *dict, int forcemask);
 
 static fz_image *
-pdf_load_jpx_imp(fz_context *ctx, pdf_document *doc, pdf_obj *rdb, pdf_obj *dict, fz_stream *cstm, int forcemask)
+pdf_load_jpx_imp(fz_context *ctx, pdf_document *doc, pdf_resource_stack *rdb, pdf_obj *dict, fz_stream *cstm, int forcemask)
 {
 	fz_image *image = pdf_load_jpx(ctx, doc, dict, forcemask);
 
-- 
2.51.0.925.g386e74b12e

