From cca51b1e62ac41ee33fadf2573f039f15fa2ddf3 Mon Sep 17 00:00:00 2001
Message-ID: <cca51b1e62ac41ee33fadf2573f039f15fa2ddf3.1757413653.git.mjg@fedoraproject.org>
From: Julian Smith <julian.smith@artifex.com>
Date: Tue, 9 Sep 2025 12:27:19 +0200
Subject: [PATCH] scripts/wrap/parse.py:get_args(): adjust to clang version
 differences

---
 scripts/wrap/parse.py | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/scripts/wrap/parse.py b/scripts/wrap/parse.py
index 863586cd3..c3bd1278c 100644
--- a/scripts/wrap/parse.py
+++ b/scripts/wrap/parse.py
@@ -2,6 +2,7 @@
 Support for accessing parse tree for MuPDF headers.
 '''
 
+import collections
 import os
 import sys
 import time
@@ -9,7 +10,7 @@ import time
 import jlib
 
 try:
-    import clang
+    import clang.cindex
 except ImportError as e:
     if '--venv' not in sys.argv:
         jlib.log( 'Warning, could not import clang: {e}')
@@ -488,6 +489,10 @@ class Arg:
 
 get_args_cache = dict()
 
+# get_args() needs to know how to get something hashable from a clang.cindex.Cursor.
+if clang:
+    g_cursor_is_hashable = issubclass(clang.cindex.Cursor, collections.abc.Hashable)
+
 def get_args( tu, cursor, include_fz_context=False, skip_first_alt=False, verbose=False):
     '''
     Yields Arg instance for each arg of the function at <cursor>.
@@ -510,7 +515,10 @@ def get_args( tu, cursor, include_fz_context=False, skip_first_alt=False, verbos
     #
     if verbose:
         jlib.log( '## Looking at args of {cursor.spelling=}')
-    key = tu, cursor, include_fz_context, skip_first_alt
+    if g_cursor_is_hashable:
+        key = tu, cursor, include_fz_context, skip_first_alt
+    else:
+        key = tu, cursor.location.file, cursor.location.line, include_fz_context, skip_first_alt
     ret = get_args_cache.get( key)
     if not verbose and state.state_.show_details(cursor.spelling):
         verbose = True
-- 
2.51.0.432.gc8867dcbdc

