From 65713633293b1c16eba599dec7912b0adc039d2e Mon Sep 17 00:00:00 2001
Message-ID: <65713633293b1c16eba599dec7912b0adc039d2e.1716640203.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Sat, 25 May 2024 14:29:56 +0200
Subject: [PATCH] skew: fix ARCH_HAS_SSE macro usage

---
 source/fitz/skew.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/fitz/skew.c b/source/fitz/skew.c
index f283e4973..b73bf752a 100644
--- a/source/fitz/skew.c
+++ b/source/fitz/skew.c
@@ -26,7 +26,7 @@
 #include <assert.h>
 #include <limits.h>
 
-#ifdef ARCH_HAS_SSE
+#if ARCH_HAS_SSE
 #include <emmintrin.h>
 #include <smmintrin.h>
 #endif
@@ -106,7 +106,7 @@ static int sum_c(const uint8_t *data)
 	return sum;
 }
 
-#ifdef ARCH_HAS_SSE
+#if ARCH_HAS_SSE
 static inline int sum_sse(const uint8_t *data)
 {
 	__m128i mm0, mm1, mm2, mm3, mm4, mm5;
-- 
2.45.1.445.ga646ce87ea

