From 811d25e56f6cdfa4ffbd1fcf028f2fa8f6c5b33c Mon Sep 17 00:00:00 2001
Message-ID: <811d25e56f6cdfa4ffbd1fcf028f2fa8f6c5b33c.1727518229.git.mjg@fedoraproject.org>
From: Julian Smith <julian.smith@artifex.com>
Date: Sat, 28 Sep 2024 12:10:24 +0200
Subject: [PATCH] fix PyMuPDF segfault on RHEL

---
 scripts/wrap/swig.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/scripts/wrap/swig.py b/scripts/wrap/swig.py
index 0134eafc0..82cd5ee42 100644
--- a/scripts/wrap/swig.py
+++ b/scripts/wrap/swig.py
@@ -1821,7 +1821,10 @@ def build_swig(
                     if enum_name.startswith( 'PDF_ENUM_NAME_'):
                         text += f'{enum_name} = {rename.class_("pdf_obj")}( obj_enum_to_obj( {enum_name}))\n'
 
-            for name in ('NULL', 'TRUE', 'FALSE', 'LIMIT'):
+            # 2024-09-28: important to not include PDF_LIMIT here, because
+            # pdf_drop_obj() treats all pdf_obj*'s as real pointers if they are
+            # >= PDF_LIMIT.
+            for name in ('NULL', 'TRUE', 'FALSE'):
                 text += f'PDF_{name} = {rename.class_("pdf_obj")}( obj_enum_to_obj( PDF_ENUM_{name}))\n'
 
             jlib.fs_update(text, path_out)
-- 
2.47.0.rc0.193.g5a85c1ce53

