diff --git i/source/fitz/load-jpx.c w/source/fitz/load-jpx.c
index a1c39f9f..056209a4 100644
--- i/source/fitz/load-jpx.c
+++ w/source/fitz/load-jpx.c
@@ -444,6 +444,10 @@ fz_load_jpx_info(fz_context *ctx, const unsigned char *data, size_t size, int *w
 }
 
 #else /* HAVE_LURATECH */
+#ifdef __cplusplus
+extern "C"
+{
+#endif
 
 #include <openjpeg.h>
 
@@ -924,6 +928,9 @@ fz_load_jpx_info(fz_context *ctx, const unsigned char *data, size_t size, int *w
 }
 
 #endif /* HAVE_LURATECH */
+#ifdef __cplusplus
+}
+#endif
 
 #else /* FZ_ENABLE_JPX */
 
