diff --git i/source/fitz/load-jpx.c w/source/fitz/load-jpx.c
index d798267b..359fc9ba 100644
--- i/source/fitz/load-jpx.c
+++ w/source/fitz/load-jpx.c
@@ -443,14 +443,18 @@ fz_load_jpx_info(fz_context *ctx, unsigned char *data, size_t size, int *wp, int
 
 #else /* HAVE_LURATECH */
 
+#ifdef __cplusplus
+extern "C"
+{
 #define OPJ_STATIC
 #define OPJ_HAVE_INTTYPES_H
 #if !defined(_WIN32) && !defined(_WIN64)
 #define OPJ_HAVE_STDINT_H
 #endif
 #define USE_JPIP
+#endif
 
-#include <openjpeg.h>
+#include <openjpeg-2.1/openjpeg.h>
 
 struct fz_jpxd_s
 {
@@ -920,6 +924,12 @@ fz_load_jpx_info(fz_context *ctx, unsigned char *data, size_t size, int *wp, int
 
 #endif /* HAVE_LURATECH */
 
+#ifdef __cplusplus
+
+}
+
+#endif
+
 #else /* FZ_ENABLE_JPX */
 
 fz_pixmap *
